include/zephyr/drivers/timer/ti_dmtimer.h

```diff
@@ -0,0 +1,158 @@
+/* Copyright (C) 2023 BeagleBoard.org Foundation
+ * Copyright (C) 2023 S Prashanth
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_TIMERS_TI_DMTIMER_H_
+#define ZEPHYR_DRIVERS_TIMERS_TI_DMTIMER_H_
+
+#include <zephyr/devicetree.h>
+
+#define TI_DM_TIMER_BASE_ADDR      DT_REG_ADDR(DT_INST(0, ti_am654_dmtimer))
+
+#define TI_DM_TIMER_IRQ_NUM        DT_IRQN(DT_INST(0, ti_am654_dmtimer))
+#define TI_DM_TIMER_IRQ_PRIO       DT_IRQ(DT_INST(0, ti_am654_dmtimer), priority)
+#define TI_DM_TIMER_IRQ_FLAGS      DT_IRQ(DT_INST(0, ti_am654_dmtimer), flags)
+
```

Comment:
I think these should be in the driver code rather than in the header, the `init` should be updated similar to other drivers so as to support multiple instances of timer, for a single tick timer this works but I think the same driver will need to be reused across multiple instances and I can see that most of our devices have multiple instances of the same  timer IP.

Reply:
Works for me, moving these to the driver code. Will leave only the register/bit definitions in the header. For supporting multiple instances let me try to make that work here.